source/use-cases/client-side-field-level-encryption-guide.txt

```diff
@@ -250,15 +257,42 @@ information on the features of envelope encryption and key management
 concepts, see :aws-docs:`AWS Key Management Service Concepts
 <kms/latest/developerguide/concepts.html#enveloping>`.
 
-The master key, used by the MongoDB driver to create and encrypt data
-keys, should be stored remotely in a :wikipedia:`Key Management System
-<Key_management#Key_management_system>`. The data encryption keys,
-generated and used by the MongoDB driver to encrypt and decrypt document
-fields, are stored in a key vault collection in the same database as the
-encrypted data.
+In this step, we create and store the master key, used by the MongoDB driver
+to encrypt data encryption keys, in the **Local Key Provider** which is the
+filesystem in our local development environment. We refer to this key as
+the "locally-managed master key" in this guide.
+
+The following diagram shows how the **master key** is created and stored:
+
+.. image:: /figures/CSFLE_Master_Key_Local.png
+   :alt: Diagram that describes creating the master key when using a local provider
+   
+The **data encryption keys**, generated and used by the MongoDB driver to
+encrypt and decrypt document fields, are stored in a key vault collection in 
+the same MongoDB replica set as the encrypted data. 
 
-To begin development, MedcoMD engineers generate a locally-managed master key
-with the **fully runnable code below**:
+The following diagram shows how the **data encryption keys** are created and
+stored:
+
+.. image:: /figures/CSFLE_Data_Key_Local.png
+   :alt: Diagram that describes creating the data encryption key when using a locally-managed master key
```

Comment:
Should this blurb/diagram about the data encryption keys go in the next step instead? i.e. **B. Create a Data Encryption Key**

Reply:
I initially was thinking that it would be helpful to include in the same section so that I could explain what the master key would be used for, but it does make more sense to place the diagram in the next section.